pkg/updater/service/datadog_agent.go

```diff
@@ -120,3 +137,78 @@ func StartAgentExperiment() error {
 func StopAgentExperiment() error {
 	return startUnit(agentUnit)
 }
+
+func setInstallInfo() {
+	// avoid rewriting the files if they already exist
+	if _, err := os.Stat(installInfoFile); err == nil {
+		log.Info("Install info file already exists, skipping")
+		return
+	}
+	tool, version := getToolVersion()
+	if err := writeInstallInfo(tool, version); err != nil {
+		log.Warnf("Failed to write install info: %s", err)
+	}
+	if err := writeInstallSignature(); err != nil {
+		log.Warnf("Failed to write install signature: %s", err)
+	}
+}
+
+func cleanInstallInfo() {
+	if err := os.Remove(installInfoFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install info file: %s", err)
+	}
+	if err := os.Remove(installSigFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install signature file: %s", err)
+	}
+}
+
+func getToolVersion() (string, string) {
+	tool := "unknown"
+	version := "unknown"
+	if _, err := exec.LookPath("dpkg-query"); err == nil {
+		tool = "dpkg"
+		toolVersion, err := getDpkgVersion()
+		if err == nil {
+			version = toolVersion
+		}
+	}
+	return tool, version
+}
+
+func getDpkgVersion() (string, error) {
+	cmd := exec.Command("dpkg-query", "--showformat=${Version}", "--show", "dpkg")
+	output, err := cmd.Output()
+	if err != nil {
+		log.Warnf("Failed to get dpkg version: %s", err)
+		return "", err
+	}
+	splitVersion := strings.Split(strings.TrimSpace(string(output)), ".")
+	if len(splitVersion) < 3 {
+		return "", fmt.Errorf("failed to parse dpkg version: %s", string(output))
+	}
+	return strings.Join(splitVersion[:3], "."), nil
+}
+
+func writeInstallInfo(tool, version string) error {
+	installInfoContent := fmt.Sprintf(`---
+install_method:
+  tool: %s
```

Comment:
Feels pretty unsafe to leave YAML the task to infer the type of tool and tool_version. Why didn't you do the same as the JSON serialization and use at least a map ?

Reply:
Also it looks like structs are defined already: https://github.com/DataDog/datadog-agent/tree/12bb48667633b45861254d9babbdf1f826eecb2b/pkg/util/installinfo